src/common-verifier/src/main/java/ee/ria/xroad/common/conf/globalconfextension/GlobalConfExtensions.java

```diff
@@ -37,25 +37,22 @@
  */
 @Slf4j
 public final class GlobalConfExtensions {
-
-    // Instance per thread, same way as in GlobalGonf/GlobalConfImpl.
-    // This way thread safety handling is same as in GlobalConf.
-    private static final ThreadLocal<GlobalConfExtensions> INSTANCE = new ThreadLocal<GlobalConfExtensions>() {
-        @Override
-        protected GlobalConfExtensions initialValue() {
-            return new GlobalConfExtensions();
-        }
-    };
-
     private OcspNextUpdate ocspNextUpdate;
     private OcspFetchInterval ocspFetchInterval;
+    private static GlobalConfExtensions instance = null;
```

Comment:
I think it should be kept `volatile`. It's similar case in class GlobalConf. See also https://stackoverflow.com/questions/16542402/lazy-initialization-singleton-class-with-volatile-variable